Sources/Sentry/SentryHub.m

```diff
@@ -590,37 +591,64 @@ - (void)captureEnvelope:(SentryEnvelope *)envelope
 
 - (SentryEnvelope *)updateSessionState:(SentryEnvelope *)envelope
 {
-    if ([self envelopeContainsEventWithErrorOrHigher:envelope.items]) {
+    BOOL handled = YES;
+    if ([self envelopeContainsEventWithErrorOrHigher:envelope.items wasHandled:&handled]) {
         SentrySession *currentSession = [self incrementSessionErrors];
 
         if (currentSession != nil) {
+            if (!handled) {
+                [currentSession endSessionCrashedWithTimestamp:[_currentDateProvider date]];
+
+                _session = [[SentrySession alloc] initWithReleaseName:_client.options.releaseName];
+                _session.environment = _client.options.environment;
+                [self.scope applyToSession:_session];
```

Comment:
`h`: That's not how we safely end and start a new session. I think we should call the following instead.
```
[self endSessionWithTimestamp:[_currentDateProvider date]];
[hub startSession];
```
As we also do in the session tracker https://github.com/getsentry/sentry-cocoa/blob/60dd0f5656d961f9a9d604ceaf8dd5d99efd393d/Sources/Sentry/SentrySessionTracker.m#L173-L174

Or am I missing something?

Reply:
Maybe I'm missing something, but this is what I get:
- `currentSession` is a clone of the current session, If a call `self endSessionWithTimestamp` this will not end `currentSession`. 
- If I end it before cloning it I cannot clone it.
- By calling `[hub endSession]` the session is immediately captured, but here, the session was being sent in the same envelope.
- By calling `startSession` the last session is also capture, which would duplicate the capture

Reply:
Please, take a look at the new approach.